### PR TITLE
added api db routes

### PR DESCRIPTION
**Pull request by callan-stretton**

### CODE REVIEW

File: server/api-routes.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+const router = express.Router()
+
+const db = {}
+
+router.use(bodyParser.json())
+
+router.get('/users', (req, res) => {
+  db.getUsers()
+    .then(result => res.json(result))
```
Comment (don-smith) [inline server/api-routes.js:13]: Needs a `.catch()`

File: server/api-routes.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+})
+
+router.get('/pings', (req, res) => {
+  db.getPings()
+    .then(result => res.json(result))
```
Comment (don-smith) [inline server/api-routes.js:21]: Needs a `.catch()`

File: server/api-routes.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+router.post('/users', (req, res) => {
+  const user = {
+    name: req.body.name,
+    username: req.body.username,
+    password: req.body.password,
+    photoUrl: req.body.photoUrl
+  }
+  db.saveUser(user)
+    .then(data => {
```
Comment (don-smith) [inline server/api-routes.js]: `data` is an array of created IDs. You should return a more explicit JSON object with the new ID in it.

File: server/api-routes.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+})
+
+router.post('/pings', (req, res) => {
+  const ping = {
+    senderId: req.params.senderId,
+    recipientId: req.params.recipientId,
+    imageUrl: req.body.imageUrl
+  }
+  db.saveUser(ping)
+    .then(data => {
```
Comment (don-smith) [inline server/api-routes.js]: `data` is an array of created IDs. You should return a more explicit JSON object with the new ID in it.

File: server/api-routes.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  db.saveUser(user)
+    .then(data => {
+      res.json(data)
+    })
+})
+
+router.post('/pings', (req, res) => {
+  const ping = {
+    senderId: req.params.senderId,
+    recipientId: req.params.recipientId,
```
Comment (don-smith) [inline server/api-routes.js]: The `senderId` and `recipientId` are not sent in on `params` but rather `req.body`.

**Review by don-smith — CHANGES_REQUESTED** [review]

File: server/api-routes.js (hunk 5 above)
Comment (callan-stretton) [reply server/api-routes.js]: all done :)

File: server/api-routes.js (hunk 2 above)
Comment (callan-stretton) [reply server/api-routes.js:21]: done :)

File: server/api-routes.js (hunk 1 above)
Comment (callan-stretton) [reply server/api-routes.js:13]: done :)

**Pull request merged**